### PR TITLE
[FIX] web: Correctly unpack groups data in kanban

### PR DESCRIPTION
**Pull request by Arcasias**
1) This PR aims to make the behavior of the mocked 'read_progress_bar'
requests closer to their server implementation.
Namely, the returned dictionnary is supposed to contains the display
names of the current groups in its keys. This was not the case and has
been fixed.

2) Before this PR, the groups returned by `web_read_group` were
unpacked using the current 'group by' field name, and not the entire
'group by' value (potentially including a granularity for date/datetime
fields). This caused mismatches in some cases and the value couldn't be
retrieved.

Also, a needless 'sum_field' parameter was passed to the
'read_progres_bar' calls, which does not exist in the request API.

This PR fixes the issues above and removed the excess parameter.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/views/kanban_view_tests.js [hunk 1]
```diff
@@ -4666,6 +4666,80 @@ QUnit.module("Views", (hooks) => {
         }
     );
 
+    QUnit.test("Move record in grouped by date with progress bars", async (assert) => {
```
Comment (Williambraecky) [inline addons/web/static/tests/views/kanban_view_tests.js:4669]: Running this test in debug shows that the values of the `sum_field` in the progress bar are wrong (always 0).
I actually have a fix for this [here](https://github.com/odoo/odoo/pull/98837/commits/4c7356a6f95c84ff2bff3167dd539e855b6d0ff4)

File: addons/web/static/src/views/relational_model.js [hunk 2]
```diff
@@ -2432,6 +2429,20 @@ export class DynamicGroupList extends DynamicList {
         };
         return new FormArchParser().parse(quickCreateForm.arch, models, this.modelName);
     }
+
+    /**
+     * @param {Object} groupData
+     * @returns {DateTime | false}
+     */
+    _parseDateValue(groupData) {
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model.js:2437]: not sure about this function name, it sounds very generic, but as far as I understand, the code is really specific

**Review by aab-odoo — APPROVED** [review]
Besides a small remark, it's ok

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Pull request closed without merging**